appserver/pom.xml

```diff
@@ -541,11 +541,17 @@
                 <groupId>javax.enterprise</groupId>
                 <artifactId>cdi-api</artifactId>
                 <version>${cdi-api.version}</version>
+                <exclusions>
+                  <exclusion>
+                      <groupId>javax.inject</groupId>
+                      <artifactId>javax.inject</artifactId>
+                  </exclusion>
+              </exclusions>
             </dependency>
             <dependency>
-                <groupId>javax.inject</groupId>
-                <artifactId>javax.inject</artifactId>
-                <version>${javax.inject.version}</version>
+                <groupId>org.glassfish.hk2.external</groupId>
```

Comment:
the hk2 bom contains dependency management for this and is already imported in nucleus/parent/pom.xml.
Removing the old javax.inject should be enough.

Reply:
Cannot find a reference to any inject in [nucleus/parent/pom.xml](https://github.com/eclipse-ee4j/glassfish/blob/EE4J_8/nucleus/parent/pom.xml)

Reply:
It's in the HK2 bom, see https://oss.sonatype.org/content/repositories/staging/org/glassfish/hk2/hk2-bom/2.5.0/hk2-bom-2.5.0.pom

It's imported here https://github.com/eclipse-ee4j/glassfish/blob/eb50c996f9d1a3b0bd8feea7b317a75ce72963eb/nucleus/parent/pom.xml#L583

In order to see the dependencyManagement you'd have to look at the effective pom.
Something like this should show it: `cd nucleus/parent ; mvn help:effective-pom`

Reply:
Removed